doc/api/http.md

```diff
@@ -901,6 +901,10 @@ also be accessed at `request.connection`.
 This event can also be explicitly emitted by users to inject connections
 into the HTTP server. In that case, any [`Duplex`][] stream can be passed.
 
+If `socket.setTimeout()` is called here, note that the timeout will
+be replaced with `server.keepAliveTimeout` when the socket has served
+a request (if `server.keepAliveTimeout` is non-zero).
```

Comment:
I think the actual logic is a bit more nuanced, if we aim for correctness: https://github.com/nodejs/node/blob/66f08fdf8bfef2a20aaded049763655f048169ef/lib/_http_server.js#L601. I would need to investigate, but my bet is that we are setting it to `0`  first to force a reset of the timout itself.

Reply:
Then we can see if we could remove that setTimeout?

Reply:
Yes, but it's out of the scope of this PR imho which is only documentation clarification.